jsmn_stream_utils.c

```diff
@@ -1032,13 +1034,17 @@ static void get_object_token_containing_kv_primitive_callback(const char *value,
 
     // do not update token, we want the parent object
     
-    if (arg->found_key == true)
+    if (arg->found_key == true && arg->token_parser->index >= arg->start_index)
```

Comment:
so the issue we were seeing was that we were looking for a specific value for the key/value pair, and without this bound, if the module found the expected value before the key, that would be returned?

I guess this solves the problem by returning the first match after the key. Should we limit the value index to be within some bounds of the key index, so that we don't find a random matching token way down the file?

Reply:
Its more of a combination
This code here prevents from every value being evaluated. i.e. if it found key "id", check if the value pair matches, if not you don't have the right key.
`        else
        {
            arg->found_key = false;
        }`
        
        and this piece of code basically sets the address of where the key was found so that get_object_token_containing_kv_primitive_callback starts at the key index.
        `            arg->token_parser->index = arg->token_parser->index + key_length;
            arg->start_index = arg->token_parser->index;`

Reply:
If you've tested to make sure it only checks the next value, sounds good to me!

Reply:
Yes I ran it trough the debugger step by step to be sure